python_modules/dagster/dagster/_core/definitions/unresolved_asset_job_definition.py

```diff
@@ -138,18 +139,26 @@ def resolve(
         assets: Sequence["AssetsDefinition"],
         source_assets: Sequence["SourceAsset"],
         default_executor_def: Optional["ExecutorDefinition"] = None,
+        asset_selection_resolver: Optional["AssetSelectionResolver"] = None,
```

Comment:
It's a little awkward to me that `asset_selection_resolver` contains kind of the same information as `assets` and `source_assets`.  E.g. it would be easy for them to get out of sync and us not notice it.

If we had the time to do a refactor, I think it might make sense to have some kind of internal `ClosedAssetGraph` class that represents an asset graph.  I.e. it would hold assets, all the source assets those assets depend on, and the . Then, instead of having `assets`, `source_assets`, and `asset_selection_resolver` arguments, it could just take a single argument.

And maybe we could use that inside `AssetLayer` as well?

That said, I realize this could be a big factor and not worth the time. Just wanted to call out that this is a little suboptimal.